docs/proposals/20191016-e2e-test-framework.md

```diff
@@ -0,0 +1,235 @@
+determined as the implementation gets more fleshed out.
+
+#### Behaviors to test
+
+* There is a Kubernetes cluster with one node that passes a healthcheck after creating a properly configured Cluster,
+  InfraCluster, Machine, InfraMachine and BootstrapConfig.
+* Creating the resources necessary for three control planes will create a three node cluster.
+* Deleting a cluster deletes all resources associated with that cluster including Machines, BootstrapConfigs,
+  InfraMachines, InfraCluster, and generated secrets.
+* Creating a cluster with one control plane and one worker node will result in a cluster with two nodes.
+* The version fields in Machines are respected within the bounds of the Kubernetes skew policy.
+* Creating a control plane machine and a MachineDeployment with two replicas will create a three node cluster with one
+  control plane node and two worker nodes.
+* Modifying a replica count on a MachineDeployment will modify the number of worker nodes and Machines in running state
+  the cluster has.
+* Optionally, Machines report failures when their underlying InfraMachines report failures.
+* Manage multiple workload clusters in different namespaces.
+* Workload Clusters created pass Kubernetes Conformance.
+
+### Risks and Mitigations
+
+These tests have two main risks: False positives/negatives and long-term technical debt. 
+
+We want these E2E tests to be useful for providers to use in a test-infra prow job. Depending on the providers’ choices,
+these tests will be able to be run as a regular test suite. Therefore they can run as pre jobs, post jobs, periodics or
+any other type of job that Prow supports. *Ideally these tests will replace, at least partially, the need for manual
+testing*. If the tests are too brittle, development velocity could take a hit, and if they’re too lenient, bugs are
+more likely to make it into production.
+
+To keep these tests healthy and useful, cluster-api will need clear ownership on this common testing infrastructure.
+Keeping these tests happy and healthy across providers will be key to success, not just now, but going forward as well.
+
+The Cluster API project is developing rapidly, and any testing framework must match that pace. The framework will need
+to be flexible enough to handle v1alpha2, 3, and onward; including beta and general releases. If the test framework is
+too restrictive, then the tests could become less useful over time, or worse, constrain development of Cluster API
+itself. The framework needs to be one of the first things ported to a new release of Cluster API, or it will be much
+more difficult to validate providers against those new releases. Clear ownership is necessary here, as is committing to
+the test framework as a release artifact.
+
+## Alternatives
+
+The `Alternatives` section is used to highlight and record other possible approaches to delivering the value proposed by
```

Comment:
I don't see any Alternatives listed here, would it be worthwhile to call that out explicitly or is there another section somewhere that should be referenced for that information?